lib/content_scripts/website/pages/series/series.js

```diff
@@ -225,3 +234,68 @@ class MarkAsWatchedNotWatched {
     }, MarkAsWatchedNotWatched.REFRESH_TIMEOUT);
   }
 }
+
+class SeriesAnimeListLinks {
+  container;
+
+  seasonsSelect;
+  seasonNameContainer;
+
+  constructor() {
+    this.seasonsSelect = document.querySelector('.seasons-select');
+    if (this.seasonsSelect) {
+      this.seasonNameContainer = this.seasonsSelect.querySelector('h4');
+      if (this.seasonNameContainer) {
+        this.createButtons()
+      } else {
+        new MutationObserver((_, observer) => {
+          this.seasonNameContainer = this.seasonsSelect.querySelector('h4');
+          if (!this.seasonNameContainer) return;
+          observer.disconnect();
+          this.createButtons();
+        }).observe(document.getElementById('content'), {
+          childList: true,
+          subtree: true,
+        })
+      }
+    } else {
+      new MutationObserver((_, observer) => {
+        this.seasonsSelect = document.querySelector('.seasons-select');
+        if (!this.seasonsSelect) return;
+        observer.disconnect();
+        new MutationObserver((_, observer) => {
+          this.seasonNameContainer = this.seasonsSelect.querySelector('h4');
+          if (!this.seasonNameContainer) return;
+          observer.disconnect();
+          this.createButtons();
+        }).observe(document.getElementById('content'), {
+          childList: true,
+          subtree: true,
+        })
+      }).observe(document.getElementById('content'), {
+        childList: true,
+        subtree: true,
+      })
+    }
```

Comment:
Callback hell. I thought of adding a general `MutationObserver` wrapper with `async` but this would be a bit too much change for this PR (and I don't know if you'd like it)
<details>
<summary></summary>

![giphy](https://github.com/ThomasTavernier/Improve-Crunchyroll/assets/63594396/66bee043-86c7-4268-9785-37dc7f35eb66)

</details>

Reply:
Thanks for considering! Let's keep the focus on this PR for now to maintain efficiency, but I'd definitely be interested in exploring that enhancement later on. Appreciate your work!